include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
> I'm starting to think that it's just better to keep this in VecGeom. 

Are you claiming that the concept 'return early if there if few lanes, aka scalar mode, unless we are using cuda because in this case there is an implicit number of lanes that we don't know (at least yet) how to count' is a concept that will be useful only for VecGeom?  What is special about VecGeom for that case?